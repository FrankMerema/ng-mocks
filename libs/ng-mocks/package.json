{
  "name": "ng-mocks",
  "version": "0.0.0",
  "description": "An Angular testing library for creating mock services, components, directives, pipes and modules in unit tests, which includes shallow rendering, precise stubs to dump child dependencies, supports Angular 5 6 7 8 9 10 11 12, jasmine and jest.",
  "keywords": [
    "angular",
    "test",
    "testing",
    "mock",
    "mocking",
    "component",
    "directive",
    "pipe",
    "provider",
    "service",
    "TestBed",
    "how-to",
    "frontend",
    "front-end",
    "jest",
    "jasmine",
    "enzyme",
    "create a dummy stub out of child dependency",
    "stubbing via dumb render"
  ],
  "author": {
    "name": "Isaac Datlof",
    "email": "ike18t@gmail.com",
    "url": "https://github.com/ike18t"
  },
  "contributors": [
    {
      "name": "Isaac Datlof",
      "email": "ike18t@gmail.com",
      "url": "https://github.com/ike18t"
    },
    {
      "name": "MG.",
      "email": "m@sudo.eu",
      "url": "https://sudo.eu"
    }
  ],
  "maintainers": [
    {
      "name": "Isaac Datlof",
      "email": "ike18t@gmail.com",
      "url": "https://github.com/ike18t"
    },
    {
      "name": "MG.",
      "email": "m@sudo.eu",
      "url": "https://sudo.eu"
    }
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/ike18t/ng-mocks"
  },
  "bugs": {
    "url": "https://github.com/ike18t/ng-mocks/issues/new"
  },
  "homepage": "https://ng-mocks.sudo.eu",
  "license": "MIT",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "files": [
    "examples",
    "migrations.json",
    "dist"
  ],
  "ng-update": {
    "migrations": "./migrations.json"
  },
  "standard": {},
  "scripts": {
    "test": "karma start"
  },
  "peerDependencies": {
    "@angular/common": ">=5.x <=12.x",
    "@angular/compiler": ">=5.x <=12.x",
    "@angular/core": ">=5.x <=12.x",
    "@angular/platform-browser": ">=5.x <=12.x"
  }
}
