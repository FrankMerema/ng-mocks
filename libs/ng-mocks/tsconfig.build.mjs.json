{
  "extends": "./tsconfig.build.cjs.json",
  "compilerOptions": {
    "module": "ES2015",
    "target": "ES2015"
  }
}
