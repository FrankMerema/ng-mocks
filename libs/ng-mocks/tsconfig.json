{
  "extends": "./tsconfig.build.json",
  "compilerOptions": {
    "experimentalDecorators": true
  },
  "include": ["./src/lib/**/*.ts"]
}
