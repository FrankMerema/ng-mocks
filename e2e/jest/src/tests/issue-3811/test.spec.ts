import { Component, Input } from '@angular/core';
import { MockBuilder, MockRender } from 'ng-mocks';

@Component({
  selector: 'hello',
  template: `<h1>Hello {{ name }}!</h1>`,
})
export class HelloComponent {
  @Input() public name: string | null = null;
}

// @see https://github.com/help-me-mom/ng-mocks/issues/3811
// The problem is that Angular adds `__ngContext__` on the host element (<mock-render>).
// It's an index of the current context: 0, 1, 2, 3.
// So if you have 2 tests, you have 2 contexts, but if you run only 1 test, you have 1 context,
// and, therefore, the snapshot assertion will fail, because the single run has `0` index, whereas in the suite run,
// it can be any other number.
// The solution is to remove __ngContext__ from the snapshot html, so there is no number with the index in it.
// The test below asserts that snapshots don't have __ngContext__ anymore.
describe('issue-3811', () => {
  beforeEach(() => {
    return MockBuilder(HelloComponent);
  });

  it('should render mock-render without __ngContext__ on the first run', () => {
    const fixture = MockRender(HelloComponent, { name: 'Joe' });
    expect(fixture).toMatchSnapshot();
  });

  it('should render mock-render without __ngContext__ on the second run', () => {
    const fixture = MockRender(HelloComponent, { name: 'Jane' });
    expect(fixture).toMatchSnapshot();
  });
});
