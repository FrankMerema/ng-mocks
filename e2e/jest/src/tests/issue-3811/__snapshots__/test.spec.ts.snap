// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`issue-3811 should render mock-render without __ngContext__ on the first run 1`] = `
<mock-render>
  <hello>
    <h1>
      Hello Joe!
    </h1>
  </hello>
</mock-render>
`;

exports[`issue-3811 should render mock-render without __ngContext__ on the second run 1`] = `
<mock-render>
  <hello>
    <h1>
      Hello Jane!
    </h1>
  </hello>
</mock-render>
`;
